message/middleware.go

```diff
@@ -0,0 +1,29 @@
+package message
+
+import "github.com/pkg/errors"
+
+// ErrMiddlewareMissingReturn is the error used when a middleware did not return either a message or an error but one is expected.
+var ErrMiddlewareMissingReturn = errors.New("a middleware did not return either a message or an error but one is expected")
+
+// Middleware is a function applied to a Message and returns the same Message after processing it or modifying it.
+// If a middleware does not return a Message, the message should be considered as discarded.
+type Middleware func(m *Message) (*Message, error)
+
+// Chain calls all given middlewares in a pipeline-fashioned process. Order of middlewares is preserved during execution.
+func Chain(middlewares ...Middleware) Middleware {
```

Comment:
Was close to being named `Pipeline` but then thought `Middleware` should be called `Pipe` and that's not exactly what it is. Also, `Middleware` is a very well-known term so I won't change that at all.

Any suggestion is welcome.

Reply:
Chain is good name :) You can also add `Middlewares*` prefix to be more explicit - `MiddlewaresChain`

Reply:
Good Point. I'm gonna rename it to `MiddlewaresChain` as suggested 👍

Reply:
I wonder if that is a good name for this function. As I see, middlewares are executed sequentially, and usually middlewares (in JS, maybe also in other languages) should have as argument `next` i.e. next middleware - middleware should determine whether to execute next middleware or break the chain.

Reply:
You are totally right. In fact what I named as Middleware is not Middleware but Handler. Middleware will just get a Handler as input, and Output another Handler.

Reply:
Let's put this PR on-hold as I'm gonna evaluate to use Watermill.io -> https://github.com/asyncapi/event-gateway/issues/43 as It has all we need (routers, handlers and middlewares, publishers, subscribers, etc) and most probably we won't need most of the code in here anymore.

Reply:
To not block you, we can go with this implementation. Please only add comment about this "problem" ot change name to HandlersChain, or something like that.

Reply:
@magicmatatjahu Middleware has been renamed to Handler, and all related accordingly to what we discussed in here.

Reply:
Thanks!